tests/commands/test_store_commands.py

```diff
@@ -370,3 +372,169 @@ def test_upload_discover_default_no_charm_file(tmp_path, monkeypatch):
         "the --charm-file option.".format(str(tmp_path / 'testcharm.charm')))
+
+
+# -- tests for list revisions command
+
+
+def test_revisions_simple(caplog, store_mock):
+    """Happy path of one result from the Store."""
+    caplog.set_level(logging.INFO, logger="charmcraft.commands")
+
+    store_response = [
+        Revision(
+            revision=1, version='v1', created_at=datetime.datetime(2020, 7, 3, 20, 30, 40),
+            status='accepted', errors=[]),
+    ]
+    store_mock.list_revisions.return_value = store_response
+
+    args = Namespace(name='testcharm')
+    ListRevisionsCommand('group').run(args)
+
+    assert store_mock.mock_calls == [
+        call.list_revisions('testcharm'),
+    ]
+    expected = [
+        "Revision    Version    Created at    Status",
+        "1           v1         2020-07-03    accepted",
+    ]
+    assert expected == [rec.message for rec in caplog.records]
+
+
+def test_revisions_name_from_metadata_ok(store_mock):
+    """The charm name is retrieved succesfully from the metadata."""
+    store_mock.list_revisions.return_value = []
+    args = Namespace(name=None)
+    with patch('charmcraft.commands.store.get_name_from_metadata') as mock:
+        mock.return_value = 'test-name'
+        ListRevisionsCommand('group').run(args)
+
+    assert store_mock.mock_calls == [
+        call.list_revisions('test-name'),
+    ]
+
+
+def test_revisions_name_from_metadata_problem(store_mock):
+    """The metadata wasn't there to get the name."""
+    args = Namespace(name=None)
+    with patch('charmcraft.commands.store.get_name_from_metadata') as mock:
+        mock.return_value = None
+        with pytest.raises(CommandError) as cm:
+            ListRevisionsCommand('group').run(args)
+        assert str(cm.value) == (
+            "Can't access name in 'metadata.yaml' file. The 'revisions' command needs to "
+            "be executed in a valid project's directory, or indicate the charm name with "
+            "the --name option.")
+
+
+def test_revisions_empty(caplog, store_mock):
+    """No results from the store."""
+    caplog.set_level(logging.INFO, logger="charmcraft.commands")
+
+    store_response = []
+    store_mock.list_revisions.return_value = store_response
+
+    args = Namespace(name='testcharm')
+    ListRevisionsCommand('group').run(args)
+
+    expected = [
+        "Nothing found",
+    ]
+    assert expected == [rec.message for rec in caplog.records]
+
+
+def test_revisions_ordered_by_revision(caplog, store_mock):
+    """Results are presented ordered by revision in the table."""
+    caplog.set_level(logging.INFO, logger="charmcraft.commands")
+
+    # three Revisions with all values weirdly similar, the only difference is revision, so
+    # we really assert later that it was used for ordering
+    store_response = [
+        Revision(
+            revision=1, version='v1', created_at=datetime.datetime(2020, 7, 3, 20, 30, 40),
```

Comment:
I would've stored that timestamp in a `t0` and used that, so i didn't have to read all of them to be happy that they're the same :-)

Reply:
:+1:

Reply:
Improved!